packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -340,8 +338,8 @@ mixin SchedulerBinding on BindingBase {
   ///
   /// The preferred way to watch for changes to this value is using
   /// [WidgetsBindingObserver.didChangeAppLifecycleState].
-  AppLifecycleState get lifecycleState => _lifecycleState;
-  AppLifecycleState _lifecycleState;
+  AppLifecycleState? get lifecycleState => _lifecycleState;
```

Comment:
I would expect this one to always return a non-null value. At the time when you can read it, `_lifecycleState` has always been initialized.

Reply:
Changing this line to:

```
AppLifecycleState get lifecycleState => _lifecycleState!;
```

causes a lot of failures in tests. For instance:

```
Null check operator used on a null value
  package:flutter/src/scheduler/binding.dart 340:58   SchedulerBinding.lifecycleState
  package:flutter/src/services/binding.dart 180:9     ServicesBinding.readInitialLifecycleStateFromNativeWindow
  package:flutter/src/services/binding.dart 34:5      ServicesBinding.initInstances
  package:flutter/src/gestures/binding.dart 66:11     GestureBinding.initInstances
  package:flutter/src/semantics/binding.dart 24:11    SemanticsBinding.initInstances
  package:flutter/src/rendering/binding.dart 32:11    RendererBinding.initInstances
  package:flutter/src/painting/binding.dart 23:11     PaintingBinding.initInstances
  package:flutter/src/widgets/binding.dart 257:11     WidgetsBinding.initInstances
  package:flutter_test/src/binding.dart 302:11        TestWidgetsFlutterBinding.initInstances
  package:flutter_test/src/binding.dart 898:11        AutomatedTestWidgetsFlutterBinding.initInstances
  package:flutter/src/foundation/binding.dart 57:5    new BindingBase
  package:flutter_test/src/binding.dart               new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding
  package:flutter_test/src/binding.dart               new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding
  package:flutter_test/src/binding.dart               new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding
  package:flutter_test/src/binding.dart               new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding
  package:flutter_test/src/binding.dart               new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding&RendererBinding
  package:flutter_test/src/binding.dart               new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding&RendererBinding&PaintingBinding
  package:flutter_test/src/binding.dart               new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding&RendererBinding&PaintingBinding&WidgetsBinding
  package:flutter_test/src/binding.dart               new TestWidgetsFlutterBinding
  package:flutter_test/src/binding.dart               new AutomatedTestWidgetsFlutterBinding
  package:flutter_test/src/_binding_io.dart 25:7      ensureInitialized
  package:flutter_test/src/binding.dart 298:100       TestWidgetsFlutterBinding.ensureInitialized
  package:flutter_test/src/widget_tester.dart 123:71  testWidgets
  material/tabbed_scrollview_warp_test.dart 79:3      main
  package:flutter_goldens/flutter_goldens.dart 43:17  main
  dart:async                                          _completeOnAsyncReturn
  package:flutter_goldens/flutter_goldens.dart        FlutterLocalFileComparator.fromDefaultComparator
  dart:async                                          _completeOnAsyncReturn
  package:flutter_goldens_client/skia_client.dart     SkiaGoldClient.getExpectations
  dart:async                                          _completeOnAsyncReturn
  package:flutter_goldens_client/skia_client.dart     SkiaGoldClient.getExpectations.<fn>
```